lib/wda/webdriveragent.js

```diff
@@ -92,26 +90,9 @@ class WebDriverAgent {
 
   async cleanupObsoleteProcesses () {
     const obsoletePids = await getPIDsListeningOnPort(this.url.port,
-      (cmdLine) => (cmdLine.includes('/WebDriverAgentRunner') || cmdLine.includes('/iproxy')) &&
+      (cmdLine) => cmdLine.includes('/WebDriverAgentRunner') &&
         !cmdLine.toLowerCase().includes(this.device.udid.toLowerCase()));
 
-    if (this.realDevice) {
```

Comment:
What happens if iproxy.quit was not called before and we try to start a new session? Is there a possibility to stop the third party socket listener?

Reply:
Do you think that is necessary. I'm not a big fan cleaning old stuff

Reply:
if something like that happens then it might be impossible to start a new session unless one kills Appium process completely

Reply:
Talked in person and came to a conclusion